Homework10/Homework10/Services/PluggingInIntoASocketService.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Homework10.Enums;
+using Homework10.Models;
+using Homework10.Services.Abstractions;
+namespace Homework10.Services
+{
+	public class PluggingInAppliancesService : IPluggingInIntoASocketService
+	{
+		private readonly IElectricalApplianceService _electricalApplianceService;
+        private readonly ILoggerService _loggerService;
+        private ElectricalAppliance[] _pluggedInElectricalAppliances = new ElectricalAppliance[0];
+        private int _pluggedInApplianceNumber = 0;
+        private bool _userWantsToContinue = true;
+        private ElectricalAppliance _electricalAppliance;
+
+        public PluggingInAppliancesService(
+            IElectricalApplianceService electricalApplianceService,
+            ILoggerService loggerService
+            )
+		{
+			_electricalApplianceService = electricalApplianceService;
+            _loggerService = loggerService;
+        }
+
+        public ElectricalAppliance ChooseElectricalAppliance()
+        {
+            var _electricalAppliance = _electricalApplianceService.ChooseElectricalAppliance();
+            return _electricalAppliance;
+        }
+
+        public bool CheckIfUserWantToContinue()
+        {
+            if (_electricalAppliance == null || _electricalAppliance == default)
+            {
+                _userWantsToContinue = false;
+            }
+            return _userWantsToContinue;
+        }
+
+        public ElectricalAppliance[] PlugInElectricalAppliances()
+        {
+            var maxQuantityOfAppliances = _electricalApplianceService.GetElectricalAppliances().Length;
+            var pluggedInApplianceQuantity = 0;
+            var totalPowerConsumption = 0;
+            Console.WriteLine($"What electrical appliance do you like to plug in? " +
+                $"Maximum quantity of plugged in electrical appliances could be {maxQuantityOfAppliances} {Environment.NewLine}");
+            while (_userWantsToContinue && pluggedInApplianceQuantity < maxQuantityOfAppliances)
+            {
+                _electricalAppliance = this.ChooseElectricalAppliance();
+                var isApplianceAlreadyPluggedIn = this.IsApplianceAlreadyPluggedIn(_electricalAppliance);
+
+                if (_electricalAppliance != null && pluggedInApplianceQuantity <= maxQuantityOfAppliances && !isApplianceAlreadyPluggedIn)
+                {
```

Comment:
I see double check: pluggedInApplianceQuantity < maxQuantityOfAppliances and pluggedInApplianceQuantity <= maxQuantityOfAppliances second one is impossible as the first one will prevent condition when ==